src/wp-includes/rest-api/endpoints/class-wp-rest-block-pattern-categories-controller.php

```diff
@@ -76,16 +76,50 @@ public function get_items_permissions_check( $request ) {
 	 * Retrieves all block pattern categories.
 	 *
 	 * @since 6.0.0
+	 * @since 6.5.0 Includes user categories from the wp_pattern_category taxonomy in the response if `user` sepecified in the request `source` param.
 	 *
 	 * @param WP_REST_Request $request Full details about the request.
 	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
 	 */
 	public function get_items( $request ) {
-		$response   = array();
-		$categories = WP_Block_Pattern_Categories_Registry::get_instance()->get_all_registered();
-		foreach ( $categories as $category ) {
-			$prepared_category = $this->prepare_item_for_response( $category, $request );
-			$response[]        = $this->prepare_response_for_collection( $prepared_category );
+		$valid_query_args = array(
+			'source' => true,
+		);
+		$query_args        = array_intersect_key( $request->get_params(), $valid_query_args );
```

Comment:
What's the purpose of this code?

Reply:
This approach was copied [from here](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php#L97), and I assumed it was a way of ensuring that any non-valid query args where dropped 🤔

Reply:
Right, but in that case it's because we are forwarding those query args wholesale to another API. Here, we are just consuming a single query parameter. So we can just do `$request['source']`.